internal/services/logz/logz_tag_rule_resource.go

```diff
@@ -108,32 +108,73 @@ func resourceLogzTagRuleCreateUpdate(d *pluginsdk.ResourceData, meta interface{}
 	}
 
 	id := tagrules.NewTagRuleID(monitorId.SubscriptionId, monitorId.ResourceGroupName, monitorId.MonitorName, "default")
-	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id)
-		if err != nil {
-			if !response.WasNotFound(existing.HttpResponse) {
-				return fmt.Errorf("checking for existing %s: %+v", id, err)
-			}
-		}
+
+	existing, err := client.Get(ctx, id)
+	if err != nil {
 		if !response.WasNotFound(existing.HttpResponse) {
-			return tf.ImportAsExistsError("azurerm_logz_tag_rule", id.ID())
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
 		}
 	}
+	if !response.WasNotFound(existing.HttpResponse) {
+		return tf.ImportAsExistsError("azurerm_logz_tag_rule", id.ID())
+	}
 
 	payload := tagrules.MonitoringTagRules{
 		Properties: &tagrules.MonitoringTagRulesProperties{
-			LogRules: expandTagRuleLogRules(d),
+			LogRules: &tagrules.LogRules{
+				FilteringTags:        expandTagRuleFilteringTagArray(d.Get("tag_filter").([]interface{})),
+				SendAadLogs:          pointer.To(d.Get("send_aad_logs").(bool)),
+				SendSubscriptionLogs: pointer.To(d.Get("send_subscription_logs").(bool)),
+				SendActivityLogs:     pointer.To(d.Get("send_activity_logs").(bool)),
+			},
 		},
 	}
 
 	if _, err := client.CreateOrUpdate(ctx, id, payload); err != nil {
-		return fmt.Errorf("creating/updating %s: %+v", id, err)
+		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 	return resourceLogzTagRuleRead(d, meta)
 }
 
+func resourceLogzTagRuleUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Logz.TagRuleClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
```

Comment:
_minor_ but this can become:

```suggestion
	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
```

(similar in the Create)